From 6fd1545334f96b3dd378a27f077a6271dd51f026 Mon Sep 17 00:00:00 2001
From: Markus Armbruster <armbru@redhat.com>
Date: Thu, 16 Jun 2011 08:47:20 -0300
Subject: [RHEL6 qemu-kvm PATCH 04/23] qdev: Move declaration of qdev_init_bdrv() into qdev.h

RH-Author: Markus Armbruster <armbru@redhat.com>
Message-id: <1308214055-31837-4-git-send-email-armbru@redhat.com>
Patchwork-id: 27221
O-Subject: [PATCH RHEL-6.2 v2 03/18] qdev: Move declaration of qdev_init_bdrv() into qdev.h
Bugzilla: 627585
RH-Acked-by: Christoph Hellwig <chellwig@redhat.com>
RH-Acked-by: Luiz Capitulino <lcapitulino@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>

Signed-off-by: Markus Armbruster <armbru@redhat.com>
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
(cherry picked from commit 7b370f513002b340a383fac961dc8c6fdf39eddc)
---
 hw/pl181.c  |    1 -
 hw/qdev.h   |    2 ++
 hw/ssi-sd.c |    1 -
 sysemu.h    |    2 --
 4 files changed, 2 insertions(+), 4 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 hw/pl181.c  |    1 -
 hw/qdev.h   |    2 ++
 hw/ssi-sd.c |    1 -
 sysemu.h    |    2 --
 4 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/hw/pl181.c b/hw/pl181.c
index 7282053..e61f8ca 100644
--- a/hw/pl181.c
+++ b/hw/pl181.c
@@ -9,7 +9,6 @@
 
 #include "sysbus.h"
 #include "sd.h"
-#include "sysemu.h"
 
 //#define DEBUG_PL181 1
 
diff --git a/hw/qdev.h b/hw/qdev.h
index c926224..24072cd 100644
--- a/hw/qdev.h
+++ b/hw/qdev.h
@@ -132,6 +132,8 @@ bool qdev_machine_modified(void);
 qemu_irq qdev_get_gpio_in(DeviceState *dev, int n);
 void qdev_connect_gpio_out(DeviceState *dev, int n, qemu_irq pin);
 
+BlockDriverState *qdev_init_bdrv(DeviceState *dev, BlockInterfaceType type);
+
 BusState *qdev_get_child_bus(DeviceState *dev, const char *name);
 
 /*** Device API.  ***/
diff --git a/hw/ssi-sd.c b/hw/ssi-sd.c
index b0fed58..a1a63b2 100644
--- a/hw/ssi-sd.c
+++ b/hw/ssi-sd.c
@@ -9,7 +9,6 @@
 
 #include "ssi.h"
 #include "sd.h"
-#include "sysemu.h"
 
 //#define DEBUG_SSI_SD 1
 
diff --git a/sysemu.h b/sysemu.h
index d2e48d6..e7e9b1e 100644
--- a/sysemu.h
+++ b/sysemu.h
@@ -210,8 +210,6 @@ extern const char *drive_get_serial(BlockDriverState *bdrv);
 extern BlockInterfaceErrorAction drive_get_on_error(
     BlockDriverState *bdrv, int is_read);
 
-BlockDriverState *qdev_init_bdrv(DeviceState *dev, BlockInterfaceType type);
-
 extern QemuOpts *drive_add(const char *file, const char *fmt, ...);
 extern DriveInfo *drive_init(QemuOpts *arg, int default_to_scsi,
                              int *fatal_error);
-- 
1.7.3.2

